src/store/index.ts

```diff
@@ -527,10 +527,16 @@ export class Main extends VuexModule {
       datasetView.lastViewed = Date.now();
       this.setDatasetViewImpl(datasetView);
       const promises: Promise<any>[] = [
-        this.api.updateDatasetView(datasetView),
-        this.setSelectedDataset(datasetView.datasetId),
-        this.setSelectedConfiguration(datasetView.configurationId)
+        this.api.updateDatasetView(datasetView)
       ];
+      if (this.dataset?.id !== datasetView.datasetId) {
```

Comment:
Can't `dataset` be null ?

Reply:
It can be null
In that case, `this.dataset?.id` it is different from `datasetView.datasetId` and the dataset must be reloaded

Reply:
No `datasetView` shouldn't be `null`, at least not from the signature of `getDatasetView`
There could be an error if the dataset view for the corresponding ID doesn't exist, but the API makes the asumption that the ID is valid